clients/src/main/java/org/apache/kafka/common/requests/FetchRequest.java

```diff
@@ -59,9 +59,18 @@ public static final class PartitionData {
         public final Uuid topicId;
         public final long fetchOffset;
         public final long logStartOffset;
-        public final int maxBytes;
         public final Optional<Integer> currentLeaderEpoch;
         public final Optional<Integer> lastFetchedEpoch;
+        public int maxBytes;
```

Comment:
Doesn't seems to be a good idea, we need to change the way we create PartitionData for ShareFetch.